internal/utils/utils.go

```diff
@@ -13,6 +13,15 @@ import (
 // DepthQueryParam is used in GET requests in Cloud API
 const DepthQueryParam = int32(1)
 
+// DereferenceOrZero returns the value of a pointer or a zero value if the pointer is nil.
+func DereferenceOrZero[T any](v *T) T {
```

Comment:
Thanks, this is definitely useful. We do have something similar defined [here](https://github.com/ionos-cloud/sdk-go/blob/7ff986147d3303a39c3cd3274931d1f02f5419d8/utils.go#L39) in the go sdk, but we need to update the version of the go sdk used in crossplane. Again, thanks for the contributions.